### PR TITLE
API Pagination

### PR DESCRIPTION
**Pull request by janabremer**
I implemented API pagination, so that the API now returns pages with 25 products each.

Thank you for your Feedback! :)

### CODE REVIEW

File: backend/src/Pagination/PaginationFactory.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace App\Pagination;
+
+use Symfony\Component\HttpFoundation\Request;
+use Pagerfanta\Pagerfanta;
+use Symfony\Component\Routing\RouterInterface;
+use Pagerfanta\Adapter\ArrayAdapter;
+
+
+class PaginationFactory
```
Comment (web-devil) [inline backend/src/Pagination/PaginationFactory.php]: das ist eher ein Service als eine Factory

eine Factory würde nur 
$adapter = new ArrayAdapter($products);
        $pagerfanta = new Pagerfanta($adapter);

und dann return  $pagerfanta

File: backend/src/Pagination/PaginationFactory.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        $pagerfanta = new Pagerfanta($adapter);
+        $pagerfanta->setMaxPerPage(25);
+        $pagerfanta->setCurrentPage($page);
+
+        foreach ($pagerfanta->getCurrentPageResults() as $product) {
+            $products[] = $product;
+        }
+        $total = $pagerfanta->getNbResults();        
+    
+        $createLinkUrl = function($targetPage) use ($route, $routeParams) {
```
Comment (web-devil) [inline backend/src/Pagination/PaginationFactory.php]: diese ganze Linkgeschichte l33-l42 in einen eigenen PaginationLinkService

**Review by web-devil — CHANGES_REQUESTED** [review]

File: backend/src/Controller/ProductController.php [hunk 3]
```diff
@@ -31,17 +31,40 @@ public function getAllProducts(ProductRepository $repository, Request $request):
...
+
+        // $paginatedCollection->addLink('self', $createLinkUrl($page));
+        // $paginatedCollection->addLink('first', $createLinkUrl(1));
+        // $paginatedCollection->addLink('last', $createLinkUrl($pagerfanta->getNbPages()));
+
+        // if ($pagerfanta->hasNextPage()) {
+        //     $paginatedCollection->addLink('next', $createLinkUrl($pagerfanta->getNextPage()));
+        // }
+        // if ($pagerfanta->hasPreviousPage()) {
+        //     $paginatedCollection->addLink('prev', $createLinkUrl($pagerfanta->getPreviousPage()));
```
Comment (marrep) [inline backend/src/Controller/ProductController.php]: is there a reason for leaving the comments?

Comment (janabremer) [reply backend/src/Controller/ProductController.php]: I was refactoring the code according to the review comments from Andreas, so this was work in progress adding a factory and services... Now its all cleaned up ;)
Anyway thank you for the comment!

**Pull request merged**